### PR TITLE
Fix RetryFlow not stopping after maxRetries

### PR DESCRIPTION
**Pull request by saftacatalinmihai**
## Purpose

Fix RetryFlow not stopping after maxRetries 

## References

References https://github.com/akka/akka/issues/28425

## Changes

- the check for retryNo was pattern matching on tuple instead of any value

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
At least one pull request committer is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

**Comment by johanandren** [comment]
OK TO TEST

**Review by johanandren — APPROVED** [review]
LGTM, thanks @saftacatalinmihai

File: akka-stream/src/main/scala/akka/stream/impl/RetryFlowCoordinator.scala [hunk 1]
```diff
@@ -102,7 +102,7 @@ import scala.concurrent.duration._
               failStage(
                 new IllegalStateException(
                   s"inner flow emitted unexpected element $result; the flow must be one-in one-out"))
-            case OptionVal.Some((_, _)) if retryNo == maxRetries => pushExternal(result)
+            case OptionVal.Some(_) if retryNo == maxRetries => pushExternal(result)
```
[the remark this answers is not in the document]
Comment (ennru) [reply akka-stream/src/main/scala/akka/stream/impl/RetryFlowCoordinator.scala]: Oh, how did I manage to not cover this in tests?

**Review by ennru — APPROVED** [review]
LGTM.

**Comment by ennru** [comment]
Thank you for identifying and fixing it!

**Pull request merged**